editor/plugins/gizmos/joint_3d_gizmo_plugin.cpp

```diff
@@ -295,7 +295,14 @@ void Joint3DGizmoPlugin::incremental_update_gizmos() {
 	if (!current_gizmos.is_empty()) {
 		update_idx++;
 		update_idx = update_idx % current_gizmos.size();
-		redraw(current_gizmos.get(update_idx));
+		uint64_t i = 0;
```

Comment:
I'm wondering if keeping the iterator globally and create it when needed if it could be better when there's a lot of gizmos.
In the header file:
```cpp
uint64_t last_gizmo_count = 0;
HashSet<EditorNode3DGizmo *>::Iterator gizmos_iterator;
```

```cpp
void Joint3DGizmoPlugin::incremental_update_gizmos() {
	if (!current_gizmos.is_empty()) {
		if (last_gizmo_count != current_gizmos.size() || !gizmos_iterator) {
			gizmos_iterator = current_gizmos.begin();
			last_gizmo_count = current_gizmos.size();
		}
		redraw(*gizmos_iterator);
		++gizmos_iterator;
	}
}
```

That's just a suggestion, I don't think there could be a lot of joint nodes in a scene so your approach is maybe better?

Reply:
Yeah, this part feels like it could be improved, to avoid having to iterate back to the right position each time.  I'm not sure if it's safe to keep the iterator around long-term though, since changes to the `HashSet` could make the iterator invalid.  Since the iterator keeps a pointer to the `keys` array, a resize could leave it pointing to an old address.  The size check would probably cover most cases, but it might still be possible to insert, causing a resize, then remove inbetween calls to `incremental_update_gizmos`, though that's probably very unlikely.

Reply:
Fair point!

Reply:
Updated to keep the pointer of the last drawn gizmo, and use that to look up where it left off in the `HashSet` on the previous call.  If the old object is removed, it should restart from the beginning, otherwise it should find it wherever it currently is in the `HashSet`, and move to the next gizmo.

Reply:
I retested with your last modifications, sounds rea good to me. Thanks.